libs/vkd3d/resource.c

```diff
@@ -3109,26 +3098,40 @@ static HRESULT d3d12_resource_bind_sparse_metadata(struct d3d12_resource *resour
         goto cleanup;
     }
 
-    if ((vr = VK_CALL(vkQueueBindSparse(vk_queue, 1, &bind_info, VK_NULL_HANDLE))) < 0)
+    /* This timeline will only get signaled on the internal sparse queue */
+    sparse->init_timeline_value = ++device->sparse_init_timeline_value;
+
+    memset(&timeline_info, 0, sizeof(timeline_info));
+    timeline_info.sType = VK_STRUCTURE_TYPE_TIMELINE_SEMAPHORE_SUBMIT_INFO;
+    timeline_info.signalSemaphoreValueCount = 1;
+    timeline_info.pSignalSemaphoreValues = &sparse->init_timeline_value;
+
+    memset(&bind_info, 0, sizeof(bind_info));
+    bind_info.sType = VK_STRUCTURE_TYPE_BIND_SPARSE_INFO;
+    bind_info.pNext = &timeline_info;
+    bind_info.imageOpaqueBindCount = 1;
+    bind_info.pImageOpaqueBinds = &opaque_bind;
+    bind_info.signalSemaphoreCount = 1;
+    bind_info.pSignalSemaphores = &device->sparse_init_timeline;
+
+    vr = VK_CALL(vkQueueBindSparse(vk_queue, 1, &bind_info, VK_NULL_HANDLE));
```

Comment:
We still need to wait for the previous timeline value, since sparse updates are unordered, and we still risk signaling N + 1 before N is signaled in theory. (Confirmed this with Tobias).